phoenix-core/src/main/java/org/apache/phoenix/compile/ProjectionCompiler.java

```diff
@@ -226,6 +226,7 @@ private static void projectAllIndexColumns(StatementContext context, TableRef ta
             String indexColName = IndexUtil.getIndexColumnName(tableColumn);
             PColumn indexColumn = null;
             ColumnRef ref = null;
+            boolean localIndexResolved = true;
```

Comment:
This should be `false`? Otherwise, this logic in catch block will never be true:
```
        if(!localIndexResolved) {
            throw e;
        }
```

Reply:
We want to swallow the exception when we have just created a new LocalIndexDataColumnRef column, as the current resolver won't recogise it.
We only want to throw an expression if we didn't create a new LocalIndexDataColumnRef, and the resolver fails for some other reason.
(That may be a non-existing case, but I wanted to make sure that I don't change the existing behaviour.)